community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java

```diff
@@ -70,24 +69,12 @@ public LogEntry readLogEntry( SOURCE channel ) throws IOException
             {
                 channel.getCurrentPosition( positionMarker );
 
-                byte typeCode;
-                byte versionCode;
-                try
+                byte versionCode = channel.get();
+                if ( versionCode == 0 )
                 {
-                    typeCode = channel.get();
-                    versionCode = 0;
-                    if ( typeCode < 0 )
-                    {
-                        // if the read type is negative than it is actually the log entry version
-                        // so we need to read an extra byte which will contain the type
-                        versionCode = typeCode;
-                        typeCode = channel.get();
-                    }
-                }
-                catch ( ReadPastEndException e )
-                {   // Make these exceptions slip by straight out to the outer handler
-                    throw e;
+                    return null;
```

Comment:
Did this method previously also explicitly return `null` on `versionCode` `0`? I can't see it straight away by looking at the existing code.